src/celeritas/ext/ScopedRootErrorHandler.cc

```diff
@@ -18,6 +18,9 @@ namespace celeritas
 {
 namespace
 {
+//---------------------------------------------------------------------------//
+bool has_root_errored{false};
```

Comment:
We may want to consider a naming convention to express that a variable is a global (to visually distinguish from local variable and data members).

Reply:
Yeah, not a bad idea. Maybe since it's an anonymous namespace (effectively static variable) we should append a `_` like we do with private data? Since it's declared and defined in the same translation unit that it's used, we at least don't have to worry about "searching" for it.

Reply:
Actually starting with underscore is reserved for the compiler.  So we need something else, eg. 'g_has_root_errored` or `static_has_root_errored` or `global_has_root_errored`, etc..

Reply:
https://en.cppreference.com/w/cpp/language/identifiers:
```
Identifiers that appear as appearing as a token or preprocessing token (i.e., not in user-defined-string-literal like operator ""id) (since C++11) of one of the following forms are reserved:

    identifiers with a double underscore anywhere;
    identifiers that begin with an underscore followed by an uppercase letter;
    in the global namespace, identifiers that begin with an underscore. 
```

Reply:
Yeah, *prefixes* are forbidden, *suffixes* are OK.

Reply:
For our project, underscore suffix means private data. A static linkage global variable is effectively private data for the translation unit.

I don't want to go 90s style and prefix data with `g`, `k`, `m`, etc. based on its linkage/scope. I don't think it'll be confusing to new contributors because it's only used as an implementation detail for functions that need to manipulate global system states (ROOT interfaces, system signals).

Reply:
> A static linkage global variable is effectively private data for the translation unit.

Unless we *always* use `this->data_` there is then no distinction between:
```
   something_ = ....; // affecting global state
   ortherother_ = ...; // affecting only the current object.
```
affecting the local or global state or am I still missing something?

Reply:
I think you're looking at this the wrong way. The `_` is about access control, not whether it's an object or a global value. We have plenty of classes that *don't* use `_` for member data but they're still object values.

If you see a `_` suffixed member in the code, you *know* that it's either in the header corresponding to the current file, or in the current file itself. So it's not hard to find, and it means it's something your file/class has complete control over.

Reply:
Added `g_` prefix for scope and `_` suffix for access control 🤪